soc/arm/ti_simplelink/cc13x2_cc26x2/power.c

```diff
@@ -184,57 +208,5 @@ void PowerCC26XX_schedulerRestore(void)
 	 */
 }
 
-#ifdef CONFIG_PM
-/* Constraint API hooks */
-
-void pm_constraint_set(enum pm_state state)
-{
-	switch (state) {
-	case PM_STATE_RUNTIME_IDLE:
-		Power_setConstraint(PowerCC26XX_DISALLOW_IDLE);
-		break;
-	case PM_STATE_STANDBY:
-		Power_setConstraint(PowerCC26XX_DISALLOW_STANDBY);
-		break;
-	default:
-		break;
-	}
-}
-
-void pm_constraint_release(enum pm_state state)
-{
-	switch (state) {
-	case PM_STATE_RUNTIME_IDLE:
-		Power_releaseConstraint(PowerCC26XX_DISALLOW_IDLE);
-		break;
-	case PM_STATE_STANDBY:
-		Power_releaseConstraint(PowerCC26XX_DISALLOW_STANDBY);
-		break;
-	default:
-		break;
-	}
-}
-
-bool pm_constraint_get(enum pm_state state)
-{
-	bool ret = true;
-	uint32_t constraints;
-
-	constraints = Power_getConstraintMask();
-	switch (state) {
-	case PM_STATE_RUNTIME_IDLE:
-		ret = (constraints & (1 << PowerCC26XX_DISALLOW_IDLE)) == 0;
-		break;
-	case PM_STATE_STANDBY:
-		ret = (constraints & (1 << PowerCC26XX_DISALLOW_STANDBY)) == 0;
-		break;
-	default:
-		break;
-	}
-
-	return ret;
-}
-#endif /* CONFIG_PM */
-
```

Comment:
As I've mentioned I think that this change is risky. Removing the capability of a target implement their own constraint and just let them be notified when a constraint is set or release is not enough if the target set constraints not using Zephyr's API. Taking as example the TI HAL, In https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c#L156-L168 we call `Power_Init()` that sets a constraint using the HAL API https://github.com/zephyrproject-rtos/hal_ti/blob/master/simplelink/source/ti/drivers/power/PowerCC26X2.c#L366-L414